### ISSUE TITLE
Exception when adding Objects by using the context-menu in BlueJ

### ISSUE THREAD

**Issue by mikebarkmin** [openpatch/scratch-for-java#5]
```
java.util.ConcurrentModificationException
	at java.base/java.util.ArrayList$Itr.checkForComodification(ArrayList.java:1043)
	at java.base/java.util.ArrayList$Itr.next(ArrayList.java:997)
	at eu.barkmin.processing.scratch.ScratchStage.draw(Unknown Source)
	at jdk.internal.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at processing.core.PApplet$RegisteredMethods.handle(PApplet.java:1387)
	at processing.core.PApplet.handleMethods(PApplet.java:1541)
	at processing.core.PApplet.handleDraw(PApplet.java:2214)
	at processing.awt.PSurfaceAWT$10.callDraw(PSurfaceAWT.java:1422)
	at processing.core.PSurfaceNone$AnimationThread.run(PSurfaceNone.java:354)
```

**Comment by mikebarkmin** [comment]
We probably need to use a `CopyOnWriteArrayList` instead of a normal ArrayList.

**Comment by mikebarkmin** [comment]
or we need to be more carefull when modifying the ArrayList. See https://www.java67.com/2015/10/how-to-solve-concurrentmodificationexception-in-java-arraylist.html

**Comment by mikebarkmin** [comment]
Fixed in 1aba9348037bfcec26d84715cd1a48f833fd2f0e
